### PR TITLE
Add all icon variants

### PR DESCRIPTION
**Pull request by stplasim**
This PR extends the icon generator script to generate all variants of all icons. The script was also improved to handle better the lage amount of icons. 

Related [ticket](https://app.clickup.com/t/2weu47d)

### CODE REVIEW

**Comment by stplasim** [comment]
Hi @alexlanz, I push the icons after the review

File: generator.js [hunk 1]
```diff
@@ -93,23 +65,46 @@ async function getSVGFile(icon, version) {
...
 ;(async () => {
-  // Setup source folder
-  setup()
+  generateIconPropsFile()
+
+  const res = await axios.get(GOOGLE_FONTS_URL)
 
-  // Get icon information from google fonts
-  const res = await axios.get(iconURL)
   // remove )]}' from the response
```
Comment (alexlanz) [inline generator.js]: What is going on here?

File: generator.js [hunk 2]
```diff
@@ -93,23 +65,46 @@ async function getSVGFile(icon, version) {
...
-  // Parse response from json to js object
   const icons = await JSON.parse(data)
 
-  // Generate icon components and Index
-  await generateComponent(icons.icons)
+  const iconChunks = new Array(
+    Math.ceil(icons.icons.length / DOWNLOAD_CHUNK_SIZE)
+  )
+    .fill()
+    .map(() => icons.icons.splice(0, DOWNLOAD_CHUNK_SIZE))
```
Comment (alexlanz) [inline generator.js]: Why do you do the split up into multiple chunks? I don't really see any benefit here how you handle it.

File: generator.js [hunk 3]
```diff
@@ -2,45 +2,39 @@ const fs = require('fs')
 const path = require('path')
 const axios = require('axios')
 
-// Base url for icons
-const iconURL = 'https://fonts.google.com/metadata/icons'
+// This is the number of icons each thread will be downloading
+const DOWNLOAD_CHUNK_SIZE = 500
+const GOOGLE_FONTS_URL = 'https://fonts.google.com/metadata/icons'
+const ICON_FAMILIES = [
+  { id: 'materialicons', namePostfix: '' },
```
Comment (alexlanz) [inline generator.js]:
```suggestion
  { id: 'materialicons', postfix: '' },
```

File: generator.js [hunk 4]
```diff
@@ -2,45 +2,39 @@ const fs = require('fs')
...
 
-/**
- * Format name.
- * Covert from snake case to camel case and prevent numbers at the beginning
- *
- * @param string - String to format
- * @returns {string} - Formatted string
- */
-function formatName(string) {
+function formatName(string, familyName) {
```
Comment (alexlanz) [inline generator.js]:
```suggestion
function formatName(string, familyPostfix) {
```

File: generator.js [hunk 5]
```diff
@@ -93,23 +65,46 @@ async function getSVGFile(icon, version) {
...
+}
+
+async function generateComponent(icons) {
+  for (let i = 0; i < icons.length; i++) {
+    const icon = icons[i]
+
+    for (let j = 0; j < ICON_FAMILIES.length; j++) {
+      const family = ICON_FAMILIES[j]
+
+      const name = formatName(icon.name, family.namePostfix)
```
Comment (alexlanz) [inline generator.js]:
```suggestion
      const name = formatName(icon.name, family.postfix)
```

File: generator.js (hunk 2 above)
Comment (stplasim) [reply generator.js]: I have cleaned it up

File: generator.js [hunk 6]
```diff
@@ -2,86 +2,79 @@ const fs = require('fs')
...
+  // remove )]}' from the response
+  const data = res.data.substring(4)
+  const icons = await JSON.parse(data)
+
+  for (let i = 0; i < icons.icons.length; i++) {
+    generateComponents(icons.icons[i])
+  }
+})()
+
+function generateIconPropsFile() {
```
Comment (alexlanz) [inline generator.js]:
```suggestion
function generatePropsFile() {
```

File: generator.js [hunk 7]
```diff
@@ -2,86 +2,79 @@ const fs = require('fs')
...
 
-/**
- * Format name.
- * Covert from snake case to camel case and prevent numbers at the beginning
- *
- * @param string - String to format
- * @returns {string} - Formatted string
- */
-function formatName(string) {
+async function generateComponents(icon) {
```
Comment (alexlanz) [inline generator.js]:
```suggestion
async function generateComponentsForAllFamilies(icon) {
```

**Review by alexlanz — APPROVED** [review]
Really nice, good job! I like the last refactoring. 🔥

**Pull request merged**